tests/rptest/tests/delete_records_test.py

```diff
@@ -782,3 +784,128 @@ class PartitionMoveExceptionReporter:
         status = consumer.consumer_status
         assert status.validator.invalid_reads == 0
         assert status.validator.out_of_scope_invalid_reads == 0
+
+
+class TopicPartitionOffset(NamedTuple):
+    topic: str
+    partition: int
+    offset: int
+
+
+class DeleteRecordsKafkaCliToolsTest(RedpandaTest):
+    """
+    To assert DeleteRecords has the same semantics with a Kafka client.
+    Uses kafka-delete-records.sh and rpk trim-prefix directly to assert
+    we achieve parity with Kafka
+    """
```

Comment:
nit: i'm not against doing this test still, but typically when we think of parity with Kafka we are thinking about server-side parity. In that regard, the _real_ test for parity is to spin up a kafka broker and use a single client to talk to both